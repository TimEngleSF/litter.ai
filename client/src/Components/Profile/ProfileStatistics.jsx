import React, { useState, useEffect } from 'react';
import { v4 as uuid } from 'uuid';
import Icons from '../Icons';

export default function ProfileStatistics({ user }) {
  const [userStatistics, setUserStatistics] = useState([]);

  const handleUserData = (category) => (
    <div key={uuid()}>
      <div className="profile-stats-row">
        <div className="profile-stats-left">
          <div className="profile-stats-icon-bg">
            <Icons name={category[0]} classname="profile-stats-icon" />
          </div>
          <h4>{category[0]}</h4>
        </div>
        <p className="profile-stats-number">{category[1]}</p>
      </div>
      <hr />
    </div>
  );

  // Converts data into array sorts the list in descending order
  useEffect(() => {
    const list = Object.keys(user).map((key) => [key, user[key]]);
    list.sort((a, b) => b[1] - a[1]);
    setUserStatistics(list);
  }, [user]);

  return (
    <>
      { userStatistics.map((category) => handleUserData(category)) }
    </>
  );
}
