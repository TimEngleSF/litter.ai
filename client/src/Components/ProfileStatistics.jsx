import React, { useState, useEffect } from 'react';

export default function ProfileStatistics({ user }) {
  const [userStatistics, setUserStatistics] = useState([]);

  const handleUserData = (category) => (
    <div>
      <div>{category[0]}</div>
      <div>{category[1]}</div>
    </div>
  );

  useEffect(() => {
    const list = Object.keys(user).map((key) => [key, user[key]]);
    list.sort((a, b) => b[1] - a[1]);
    setUserStatistics(list);
  }, [user]);

  return (
    <div className="profile-container">
      { userStatistics.map((category) => handleUserData(category)) }
    </div>
  );
}
