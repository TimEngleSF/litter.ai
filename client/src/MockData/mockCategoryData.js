const categoryData = {
  plastic: {
    name: 'plastic',
    category: 'Recycle',
    icon: '',
    alt: 'recycle icon',
    description: 'Not all plastic is recyclable, even if you see the chasing arrows sign!\nIf the plastic sign has a number #1 or #2 inside of the chasing arrows, it is most likely recyclable.',
    desc_long: '<p>Not all plastic is recyclable, even if you see the chasing arrows sign!</p>\n<p>If the Plastic sign has a number #1 or #2 inside of the chasing arrows, it is most likely recyclable.</p>\n<p>Sometimes, Plastic #5 is recyclable but check with your zip code to be most certain.</p>\n<p>Most other numbers (i.e. 3, 4, 6, 7) should be sent to landfill.</p>\n<p>To be recycled, plastics should be clean of food/residue.</p>\n<p>Soft plastics like plastic bags or saran wrap should be thrown in the trash.</p>\n<p>Cosmetics should only be recycled if the item is not mixed with metal, and if the cosmetic portion is removed.</p>',
  },
  metal: {
    name: 'metal',
    category: 'Recycle',
    icon: '',
    alt: 'recycle icon',
    description: 'Most types of metal are recyclable!\nAluminum/tin cans are recyclable.  Rinsing/washing them first will help their recyclability and keep your kitchen smelling fresh.\nMixed items (combination metal, plastic, or glass) should be separated into their component parts.\n',
    desc_long: '<p>Most types of metal are recyclable!</p>\n<p>Aluminum/tin cans are recyclable.  Rinsing/washing them first will help their recyclability and keep your kitchen smelling fresh.</p>\n<p>Mixed items (combination metal, plastic, or glass) should be separated into their component parts.</p>',
  },
  cardboard: {
    name: 'cardboard',
    category: 'Recycle',
    icon: '',
    alt: 'recycle icon',
    description: 'Most cardboard is recyclable.  Break cardboard down so it lays flat to save space in your bin.\nRemove any plastic from the cardboard.  If soiled (with food), it can go into compost.',
    desc_long: '<p>Cardboard is one of the easiest and most common materials to recycle!</p>\n<p>It’s best to break down cardboard so that it lays flat. This will help you save space in your bin.</p>\n<p>Remove any plastic from the cardboard before recycling.</p>\n<p>If the item is a takeout box or a pizza box:</p>\n<ul><li>If it’s soiled with food, it can go into compost</li><li>If it’s clean, it can be recycled.</li></ul>',
  },
  paper: {
    name: 'paper',
    category: 'Recycle',
    icon: '',
    alt: 'recycle icon',
    description: 'Types of paper than can be recycled: office printer paper, newspaper, and magazines, (paper towels go to the landfill)\nPaper for recycling should be clean and not soiled.\nClick the link below for more info.',
    desc_long: '<p>Many different kinds of paper can be recycled, including:</p>\n<ul><li>newspaper</li><li>colored office paper</li><li>white office paper</li><li>magazines</li><li>catalogs</li><li>phone books</li></ul>\n<p>Some type of paper that cannot be recycled are:</p>\n<ul><li>coated and treated paper</li><li>paper with food waste</li><li>juice and cereal boxes</li><li>paper cups</li><li>paper towels</li><li>paper or magazines laminated with plastic</li></ul>',
  },
  glass: {
    name: 'glass',
    category: 'Recycle',
    icon: '',
    alt: 'recycle icon',
    description: 'Most glass is recyclable, and should be separated from any non-glass part before being placed into the recycling bin.\nFor a list of glass that cannot be recycled, click the link below.',
    desc_long: '<p>Most glass is recyclable.</p>\n<p>Most food and beverage glass containers can be recycled.</p>\n<p>The glass part should be separated from the non-glass part to be recycled. Otherwise it should be sent to landfill.</p>\n<p>The following types of glass <u>cannot</u> be recycled:</p>\n<ul><li>Drinking or wine glasses and plates</li><li>Ceramics, Pyrex or other heat resistant glass</li><li>Light bulbs</li><li>Computer monitors, phone screens</li><li>Plate glass: windows, sliding doors (can be recycled separately)</li><li>Safety glass, car windshields</li><li>Art glass and leaded crystal</li><li>Mirrors</li></ul>',
  },
  compost: {
    name: 'compost',
    category: 'Compost',
    icon: '',
    alt: 'compost icon',
    description: 'In general, organic material from the kitchen or yard can be composted.  Please use a compost bag, not a plastic bag when composting.\nFor a longer list of what can and cannot be composted, click the link below.',
    desc_long: '<p>In general, organic material from the kitchen or yard can be composted.  Please use a compost bag, not a plastic bag when composting.  Here’s a list:</p>\n<h1><b>Compostable</b></h1>\n<p>Food waste from the kitchen</p>\n<ul><li>Leftovers</li><li>Trimmings (i.e banana peels)</li><li>Coffee grounds</li></ul><p>Food soiled paper</p>\n<ul><li>Coffee filters</li><li>Soiled pizza boxes</li><li>Paper bags, napkins</li><li>Be sure to remove metal or plastic from the paper</li></ul><p>Plants</p>\n<ul><li>Yard trimmings</li><li>Fallen leaves</li></ul><p>Other</p>\n<ul><li>Cotton balls</li><li>fur/hair</li><li>Wood, cork</li></ul><h1><b>Not compostable</b></h1>\n<ul><li>Aluminum foil</li><li>“Biodegradable” plastic</li><li>Cooking oil</li><li>Animal (cat/dog) droppings</li><li>Ceramic/glass dishware</li><li>Clothing</li></ul>',
  },
  trash: {
    name: 'landfill',
    category: 'Trash',
    icon: '',
    alt: 'landfill icon',
    description: 'Household items that cannot be recycled or composted should go into the landfill bin.',
    desc_long: '<p>Most household items that can’t be recycled or composted should go into the landfill bin.</p>\n<p>Some common items include:</p><ul><li>Plastic that isn’t #1, #2 or #5</li><li>Soiled plastic takeout</li><li>Ceramics</li><li>Light bulbs</li><li>Mixed plastic, metal, and glass that can’t be separated.</li></ul><p>Some items that should be disposed separately from the landfill, recycling, or compost include:</p><li>Batteries</li><li>E-waste</li><li>Large objects like furniture or mattresses</li><li>Cooking oil</li></ul>',
  },
  other: {
    name: 'other',
    category: 'Unknown',
    icon: '',
    alt: 'landfill icon',
    description: 'We recommend that you look up your local waste management service to learn how to properly dispose of these items.',
    desc_long: '<p>Some items that should be disposed separately from the landfill, recycling, or compost, such as:</p><ul><li>Batteries</li><li>E-waste such as computers or printers</li><li>Large objects like furniture or mattresses</li><li>Large sporting equipment such as bicycles</li><li>Cooking oil</li><li>Tires</li><li>Deceased animals</li></ul><p>We recommend that you look up your local waste management service to learn how to properly dispose of these items.</p>',
  },
  unknown: {
    name: 'unknown',
    category: 'Unknown',
    icon: '',
    alt: 'landfill icon',
    description: 'Some items are just hard to classify!  If you’re unsure of how to sort this item, click the link below for general guidelines.',
    desc_long: '<h1>Recyclable:</h1>\n <ul><li>#1, #2 plastic</li><ul><li>Should be clean without food or other contamination</li></ul></ul><ul><li>Most metal</li><ul><li>Aluminum or tin cans, water bottles</li><li>Should be clean without food or other contamination</li></ul></ul><ul><li>Cardboard and clean paper</li><ul><li>Should be clean without food or other contamination</li><li>Cardboard boxes, white copy paper, colored paper, newspapers, magazine</li></ul></ul><ul><li>Glass</li><ul><li>Glass jars and bottles</li></ul></ul><p>Recyclable materials should not be mixed with other materials such as metal (i.e. some cosmetics casing)--these go to landfill.</p>\n<h1>Not Recyclable</h1><ul><li>Soft plastics like bags or saran wrap</li><li>Plastics such as: #3, #4, #6, or #7</li><li>Material contaminated with lots of food</li><li>Ceramic dishware</li><li>Mixed material objects with an inseparable combination of plastic, metal, and glass such as some cosmetic casings and juice boxes</li></ul><h1>Compostable</h1>\n<ul><li>Food waste from the kitchen</li><li>Food soiled paper</li><li>Plants</li></ul><h1>Not compostable</h1>\n<ul><li>Aluminum foil</li><li>“Biodegradable” plastic</li><li>Cooking oil</li><li>Animal (cat/dog) droppings</li></ul><h1>Landfill</h1>\n<p>Most household items that can’t be recycled or composted should go into the landfill bin.</p><p>Some common items include:</p><ul><li>Plastic that isn’t #1, #2 or #5</li>Soiled plastic takeout</li><li>Ceramic such as coffee mugs or dishware</li><li>Light bulbs</li><li>Mixed plastic, metal, and glass itemsthat can’t be separated.</li></ul><h1>Other</h1>\n<p>Some items that should be disposed separately from the landfill, recycling, or compost, such as:</p><ul><li>Batteries</li><li>E-waste such as computers or printers</li><li>Large objects like furniture or bicycles</li></ul>',
  },
};

export default categoryData;
