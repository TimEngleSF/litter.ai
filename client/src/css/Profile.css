/* General classes */

.profile-container {
  width: 100%;
}

/* User logged in */

.profile-container
.profile-login-wrapper {
  height: 100dvh;
  max-height: 100%;
  width: 100%;
  display: flex;
  flex-direction: column;
  justify-content: space-between;
  align-items: center;
}

.profile-container
.profile-login-wrapper
.profile-login-header {
  width: 100%;
  background-color: var(--main-green);
  border-radius: 0 0 1.5rem 1.5rem;
}

.profile-container
.profile-login-wrapper
.profile-login-header h1 {
  margin: 5rem 0 2rem 2rem;
  color: var(--white);
  font-size: 1.7rem;
}

.profile-container
.profile-login-wrapper
.profile-logout-button {
  width: 80%;
  height: 7%;
  background-color: var(--gray);
  display: flex;
  justify-content: space-between;
  align-items: center;
  border-radius: 0.7rem;
}

.profile-container
.profile-login-wrapper
.profile-logout-button h2 {
  margin-left: 1.3rem;
  font-size: 0.8rem;
}

.profile-container
.profile-login-wrapper
.profile-logout-button button {
  height: 40%;
  width: 15%;
  margin-right: 1.2rem;
  font-size: 0.6rem;
  color: var(--main-green);
  border-radius: 10rem;
  border-width: 0.01rem;
  border-color: var(--main-green);
}

.profile-container
.profile-login-wrapper
.profile-logout-button button:hover {
  cursor: pointer;
}

.profile-container
.profile-login-wrapper
.profile-statistics h2 {
  font-size: 0.9rem;
}

/* User not logged in */

.profile-container
.profile-logout-wrapper {
  width: 100%;
  height: 100dvh;
  display: flex;
  flex-direction: column;
  justify-content: space-between;
  align-items: center;
}

.profile-container
.profile-logout-wrapper
.profile-logout-top {
  width: 100%;
  height: 100%;
}

.profile-container
.profile-logout-wrapper
.profile-logout-top
.profile-logout-bg {
  width: 100%;
  height: 30dvh;
  border-radius: 0 0 1rem 10rem;
  background-color: var(--main-green);
}

.profile-container
.profile-logout-wrapper
.profile-logout-top
.profile-logout-header {
  text-align: center;
}

.profile-container
.profile-logout-wrapper
.profile-logout-top
.profile-logout-header h1 {
  margin: 2rem 0 0 0;
}

.profile-container
.profile-logout-wrapper
.profile-logout-top
.profile-logout-header h3 {
  font-weight: 400;
  margin: 0;
}

.profile-container
.profile-logout-wrapper
.profile-logout-top
.profile-buttons {
  margin-top: 2rem;
}

.profile-container
.profile-logout-wrapper
.profile-logout-top
.profile-buttons button {
  width: 100%;
}

.profile-container
.profile-logout-wrapper
.profile-logout-top
.profile-buttons
.button-signup {
  background-color: var(--main-green-half);
}

.profile-container
.profile-logout-wrapper
.profile-logout-bottom {
  width: 100%;
  display: flex;
  flex-direction: column;
  align-items: center;
}

.profile-container
.profile-logout-wrapper
.profile-logout-bottom a {
  width: 50%;
}

.profile-container
.profile-logout-wrapper
.profile-logout-bottom
.button-home {
  width: 100%;
  max-width: 653px;
  height: 6dvh;
  margin: 1rem 0;
  border: none;
  border-radius: 10rem;
  font-size: 1.2rem;
  background-color: var(--white);
}

.profile-container
.profile-logout-wrapper
.profile-logout-bottom
.button-home:hover {
  cursor: pointer;
  background-color: var(--main-green);
  color: var(--white);
	box-shadow: 0.25em 0.25em .75em -.5em var(--shadow-color);
  font-weight: bold;
}

.profile-container
.profile-logout-wrapper
.profile-buttons
a {
  width: 50%;
}

.profile-container
.profile-logout-wrapper
.profile-buttons
button:hover {
  cursor: pointer;
  background-color: var(--main-green);
  color: var(--white);
	box-shadow: 0.25em 0.25em .75em -.5em var(--shadow-color);
  font-weight: bold;
}

/* Profile Statistics */

.profile-container
.profile-statistics {
  height: 40%;
  width: 80%;
  -ms-overflow-style: none;
  scrollbar-width: none;
  overflow-y: scroll;
}

.profile-container
.profile-statistics::-webkit-scrollbar {
  width: 0;
  background: transparent;
}

.profile-stats-row {
  display: flex;
  justify-content: space-between;
  align-items: center;
}

.profile-stats-row
.profile-stats-left {
  display: flex;
  align-items: center;
}

.profile-stats-row
.profile-stats-left h4 {
  text-transform: capitalize;
}

.profile-stats-row
.profile-stats-left
.profile-stats-icon-bg {
  width: 2rem;
  height: 2rem;
  display: flex;
  align-items: center;
  justify-content: center;
  background-color: var(--lb-gray);
  border-radius: 100rem;
}

.profile-stats-row
.profile-stats-left
.profile-stats-icon-bg
.profile-stats-icon {
  font-size: 1rem;
}

.profile-stats-row
.profile-stats-left h4 {
  margin-left: 0.5rem;
  font-size: 0.8rem;
}

.profile-stats-row p {
  margin-right: 0.5rem;
  font-size: 0.8rem;
  font-weight: bold;
}

/* Phones */
@media only screen and (max-width: 600px) {
  .profile-container {
    grid-column-start: 1;
  }

  .profile-container
  .profile-password button {
    width: 20%;
  }
}

/* Portrait tablets and large phones */
@media only screen and (min-width: 601px) and (max-width: 1024px) {
  .profile-container {
    grid-column-start: 2;
  }
}

/* Small laptops and desktops */
@media only screen and (min-width: 1025px) {
  .profile-container {
    grid-area: 1 / 2 / span 10 / span 1;
  }
}