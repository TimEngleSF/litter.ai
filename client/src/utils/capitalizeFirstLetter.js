export const capitalizeList = (list) => (
  Object.keys(list).map(
    (key) => [key.charAt(0).toUpperCase() + key.slice(1), list[key]],
  )
);

export const capitalizeWord = (word) => (
  word.charAt(0).toUpperCase() + word.slice(1)
);
