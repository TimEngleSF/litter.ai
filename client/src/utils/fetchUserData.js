/* eslint-disable consistent-return */
import Cookies from 'js-cookie';
import fetchData from './fetch';

export async function fetchLogin(body) {
  try {
    const response = await fetchData('login', 'POST', body);
    if (response.token) {
      Cookies.set('authToken', response.token, { expires: 7 });
      Cookies.set('username', response.user.displayUsername, { expires: 7 });
    }
    return response;
  } catch (err) {
    console.error(err);
  }
}

export async function fetchLogOut() {
  try {
    await fetchData('logout', 'POST');
    await Cookies.remove('authToken');
    await Cookies.remove('username');
  } catch (err) {
    console.error(err);
  }
}

export async function fetchLeaderboardData(path) {
  try {
    const response = await fetchData(path, 'GET');
    return response;
  } catch (err) {
    console.error(err);
  }
}

export async function fetchProfileData(user) {
  try {
    const response = await fetchData(`photo/${user}`, 'GET');
    return response;
  } catch (err) {
    console.error(err);
  }
}

export async function fetchRegister(body) {
  try {
    const response = await fetchData('register', 'POST', body);
    if (response.status === 'pending') {
      const loginBody = {
        email: body.email,
        password: body.password,
      };
      const loginResponse = await fetchLogin(loginBody);
      return loginResponse;
    }
    return response;
  } catch (err) {
    console.error(err);
  }
}
