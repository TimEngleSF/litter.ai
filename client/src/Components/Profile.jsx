import React from 'react';
import { Link } from 'react-router-dom';
import { userData, userPictureData } from '../MockData/mockUserData';
import '../css/Profile.css';
import ProfileStatistics from './ProfileStatistics';

export default function Profile() {
  return (
    <div className="profile-container">
      <div className="profile-header">
        <h1>{userData[0].username}</h1>
        <h3>{userData[0].email}</h3>
        <div className="profile-background" />
      </div>
      <div className="profile-password">
        <h2>Password</h2>
        <button type="button">Change</button>
      </div>
      <div className="profile-statistics">
        <h2>My Waste Statistics</h2>
        <ProfileStatistics user={userPictureData[0].pictureData} />
      </div>
      <div className="profile-buttons">
        <Link to="/capture"><button type="button">Capture Picture</button></Link>
        <Link to="/"><button className="button-home" type="button">Home</button></Link>
      </div>
    </div>
  );
}
