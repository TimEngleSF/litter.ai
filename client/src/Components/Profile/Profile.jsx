import React, { useState, useEffect } from 'react';
import { Link } from 'react-router-dom';
import { fetchProfileData, fetchLogOut } from '../../utils/fetchUserData';
import '../../css/Profile.css';
import ProfileStatistics from './ProfileStatistics';

export default function Profile({ user, setUser }) {
  const [userStatisticData, setUserStatisticData] = useState(null);

  const handleLogout = async () => {
    await fetchLogOut();
    await setUser(null);
  };

  // Authenticates user and fetches user's waste statistics
  useEffect(() => {
    if (user) {
      const fetchData = async () => {
        const userData = await fetchProfileData(user.toLowerCase());
        setUserStatisticData(userData);
      };
      fetchData();
    }
  }, [user]);

  return (
    <div className="profile-container main-container">
      { user ? (
        <div className="profile-login-wrapper">
          <div className="profile-login-header">
            <h1>{user}</h1>
          </div>
          <div className="profile-logout-button">
            <h2>User Status</h2>
            <button type="button" onClick={handleLogout}>Log out</button>
          </div>
          <div className="profile-statistics">
            <h2>My Waste Statistics</h2>
            { userStatisticData
              && <ProfileStatistics user={userStatisticData.pictureData} />}
          </div>
          <div className="profile-buttons lower-buttons">
            <Link to="/capture">
              <button type="button">
                Capture Picture
              </button>
            </Link>
            <Link to="/">
              <button className="button-home" type="button">
                Home
              </button>
            </Link>
          </div>
        </div>
      )
      // Screen if a user is not logged in
        : (
          <div className="profile-logout-wrapper">
            <div className="profile-logout-top">
              <div className="profile-logout-bg" />
              <div className="profile-logout-header">
                <h1>Join us!</h1>
                <h3>Keep track of your sorting statistics</h3>
              </div>
              <div className="profile-buttons lower-buttons">
                <Link to="/login">
                  <button type="button">
                    Log In
                  </button>
                </Link>
                <Link to="/register">
                  <button type="button" className="button-signup">
                    Sign Up
                  </button>
                </Link>
              </div>
            </div>
            <div className="profile-logout-bottom">
              <Link to="/">
                <button className="button-home" type="button">
                  Home
                </button>
              </Link>
            </div>
          </div>
        )}
    </div>
  );
}
