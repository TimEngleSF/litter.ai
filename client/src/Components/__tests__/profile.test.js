/* eslint-disable testing-library/no-unnecessary-act */
/* eslint-disable react/jsx-filename-extension */
import React from 'react';
import renderer from 'react-test-renderer';
import {
  render,
  act,
  screen,
  fireEvent,
  waitFor,
} from '@testing-library/react';
import { MemoryRouter as Router } from 'react-router-dom';
import Profile from '../Profile/Profile';
import ProfileStatistics from '../Profile/ProfileStatistics';
import * as fetchUserData from '../../utils/fetchUserData';
import { userPictureData } from '../../MockData/mockUserData';

describe('Profile component with user logged in', () => {
  const mockSetState = jest.fn();
  const profile = (
    <Router>
      <Profile user="Briana30" setUser={mockSetState} />
    </Router>
  );
  const statistics = (
    <Router>
      <ProfileStatistics user={userPictureData[0].pictureData} />
    </Router>
  );
  // Function to simulate an API call
  const mockCall = async (data) => {
    jest.spyOn(fetchUserData, 'fetchProfileData').mockResolvedValue(data);
  };

  beforeEach(async () => { await act(() => { mockCall(userPictureData); }); });
  afterEach(() => { jest.restoreAllMocks(); });

  test('Profile matches the current snapshot', async () => {
    // Snapshot currently doesn't wait for child component to render
    const tree = await act(async () => renderer.create(profile));
    expect(tree.toJSON()).toMatchSnapshot();
  });

  test('Username should appear at the top of the screen', async () => {
    await act(() => render(profile));
    expect(screen.getByText('Briana30')).toBeInTheDocument();
  });

  test('Clicking on "Log out" button should remove the user\'s data', async () => {
    jest.spyOn(fetchUserData, 'fetchLogOut');
    await act(() => render(profile));
    await act(() => fireEvent.click(screen.getByRole('button', { name: /Log out/i })));
    await waitFor(() => { expect(mockSetState).toHaveBeenCalled(); });
  });

  test('Statistics should reflect the user\'s data', async () => {
    await act(() => render(statistics));
    expect(screen.getByTestId(/paper-icon/i)).toBeInTheDocument();
  });
});
