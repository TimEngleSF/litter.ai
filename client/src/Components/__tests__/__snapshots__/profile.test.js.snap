// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Profile component with user logged in Profile matches the current snapshot 1`] = `
<div
  className="profile-container main-container"
>
  <div
    className="profile-login-wrapper"
  >
    <div
      className="profile-login-header"
    >
      <h1>
        Briana30
      </h1>
    </div>
    <div
      className="profile-logout-button"
    >
      <h2>
        User Status
      </h2>
      <button
        onClick={[Function]}
        type="button"
      >
        Log out
      </button>
    </div>
    <div
      className="profile-statistics"
    >
      <h2>
        My Waste Statistics
      </h2>
    </div>
    <div
      className="profile-buttons lower-buttons"
    >
      <a
        href="/capture"
        onClick={[Function]}
      >
        <button
          type="button"
        >
          Capture Picture
        </button>
      </a>
      <a
        href="/"
        onClick={[Function]}
      >
        <button
          className="button-home"
          type="button"
        >
          Home
        </button>
      </a>
    </div>
  </div>
</div>
`;
